### PR TITLE
Add sensitive words index

### PR DESCRIPTION
**Pull request by SteveHicksCH**
Jira: IVP-101

Add the sensitive-words index and creation script (inputs agreed with support)

The delete index script will be deleted once we stabilise our elasticsearch configuration (i.e. it is not part of any release we will put into live)

### CODE REVIEW

File: scripts/create-indices.sh [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+# set defaults
+replics=1
+shards=1
+
+while getopts 'e:r:s:h' arg
+do
+  case "$arg" in
+    e) 
+        es_url=$OPTARG         # 'chs-pp-es1.ch.gov.uk:9200'
```
Comment (jsimons1989) [inline scripts/create-indices.sh]: This is a public repo. I don't think we should have our URLs as an example.

File: scripts/create-indices.sh [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    ;;
+    r) 
+        replics=$OPTARG
```
Comment (jsimons1989) [inline scripts/create-indices.sh]: Typo on replicas

File: scripts/create-indices.sh [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+      echo "Create Sensitive Words Index Script"
+      echo ""
+
+      echo ""
+      echo "Options are:"
+      echo ""
+      echo "      OPTION    ENV-VAR        DESCRIPTION                              EXAMPLE"
+      echo "        -e      es_url         The elastic search url.                  chs-pp-es1.ch.gov.uk:9200"
+      echo "        -r      replics        The number of replicas                   3 (default = 1)"
+      echo "        -s      shards         The number of shards                     3 (default = 1)"
```
Comment (jsimons1989) [inline scripts/create-indices.sh]: The examples should match what is expected to be input from the user (The "default = 1" shouldn't be there?)

File: scripts/create-indices.sh [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    \?)
+      echo "ERROR: Unknown option $OPTARG"
+    ;;
+  esac
+done
+
+es_url=${es_url:?ERROR: var not set [-e es_url]}
+echo "elastic search url: $es_url"
+echo "number of replicas: $replics"
+echo "number of replicas: $shards"
```
Comment (jsimons1989) [inline scripts/create-indices.sh]: Wrong message in the echo. Should be "number of shards"

File: scripts/index-config/sensitive-words-index-mapping.json [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+          "properties": {
+                "master_text": {
+                    "type": "keyword"
+                },
+              "sensitive": {
+                  "analyzer": "sensitive_word_analyzer",
+                  "search_analyzer": "ngram_word_analyzer",
+                  "type": "text"
+              }
+            }
```
Comment (jsimons1989) [inline scripts/index-config/sensitive-words-index-mapping.json]: Can the formatting be fixed please

File: scripts/create-indices.sh [hunk 6]
```diff
@@ -52,4 +52,12 @@ fi
 
 sed -e "s/VAR_NUMBER_OF_REPLICAS/${replics}/;s/VAR_NUMBER_OF_SHARDS/${shards}/" ./index-config/sensitive-words-index-mapping.json > $targetscript
 
-curl -X PUT ${es_url}/sensitive-words -H "Content-Type: application/json" -d@./$targetscript
+response=$(curl -w '%{http_code}' -X PUT ${es_url}/sensitive-words -H "Content-Type: application/json" -d@./$targetscript)
+echo "Elasticsearch response = $response"
+status=${response##*\}}
+if [ "$status" -eq 400 ]
```
Comment (markwilliam5) [inline scripts/create-indices.sh]: I think I would be happier with -ne 200, what do you think?

File: scripts/create-indices.sh (hunk 1 above)
Comment (SteveHicksCH) [reply scripts/create-indices.sh]: Good point, I copied this from the Loadinator (which is private). I have put a localhost example

File: scripts/create-indices.sh (hunk 2 above)
Comment (SteveHicksCH) [reply scripts/create-indices.sh]: I have corrected this typo for my shell variable name

File: scripts/create-indices.sh (hunk 3 above)
Comment (SteveHicksCH) [reply scripts/create-indices.sh]: I have added a new column for Default value

File: scripts/create-indices.sh (hunk 4 above)
Comment (SteveHicksCH) [reply scripts/create-indices.sh]: Updated

File: scripts/create-indices.sh (hunk 6 above)
Comment (SteveHicksCH) [reply scripts/create-indices.sh]: Updated code and retested

File: scripts/index-config/sensitive-words-index-mapping.json (hunk 5 above)
Comment (SteveHicksCH) [reply scripts/index-config/sensitive-words-index-mapping.json]: Reformatted

**Pull request merged**